tests/test_install.sh

```diff
@@ -1,192 +1,284 @@
+    # kubectl delete namespace ${INGRESS_NAMESPACE}
+  else
+    echo "########################################################################################"
+    echo "#  Skipping NGINX Ingress Controller uninstall"
+    echo "########################################################################################"
   fi
 
 elif [[ "$1" == "upgrade" ]]; then
 
+  # Get the Redis password
+  export REDIS_PASSWORD=$(kubectl get secret --namespace ${REDIS_NAMESPACE} redis -o jsonpath="{.data.redis-password}" | base64 -d)
+
   echo "########################################################################################"
   echo "#  Upgrade Codesealer Release"
   echo "########################################################################################"
-
   helm repo update codesealer
-  helm upgrade codesealer ${CODESEALER_HELM_CHART} --namespace codesealer-system \
-    --set codesealerToken="${CODESEALER_TOKEN}" \
-    --set worker.ingress.namespace="${INGRESS_NAMESPACE}" \
+  helm upgrade codesealer-${RELEASE_VER} ${CODESEALER_HELM_CHART} --namespace codesealer-system \
+    --set codesealerToken=${CODESEALER_TOKEN} \
+    --set worker.ingress.namespace=${INGRESS_NAMESPACE} \
+    --set worker.ingress.deployment=${INGRESS_DEPLOYMENT} \
+    --set worker.ingress.port=${INGRESS_PORT} \
+    --set worker.redis.service.name=redis-${RELEASE_VER}-master \
+    --set worker.config.bootloader.redisUser=default \
+    --set worker.config.bootloader.redisPassword=${REDIS_PASSWORD} \
+    --set worker.config.bootloader.redisUseTLS=false \
+    --set worker.config.bootloader.redisIgnoreTLS=true \
     --set worker.config.endpoint.wafMonitorMode=false \
     --set worker.config.endpoint.enableWaf=true \
     --set worker.config.endpoint.wafFullTransaction=true \
-    --set worker.config.endpoint.paranoiaLevel=1
+    --set worker.config.endpoint.crs.paranoiaLevel=1 \
+    --wait --timeout=90s
 
   echo "########################################################################################"
   echo "#  Upgrade Codesealer"
   echo "########################################################################################"
-
   read -r -s -p 'Press any key to continue.'
   kubectl rollout restart deployments --namespace codesealer-system
-  kubectl rollout status deployments --namespace codesealer-system
+
+  echo "########################################################################################"
+  echo "#  Please wait 1 minute..."
+  echo "########################################################################################"
+  sleep 60
 
   echo "########################################################################################"
   echo "#  Upgrade Codesealer Worker Sidecar for NGINX Ingress Controller"
   echo "########################################################################################"
   read -r -s -p 'Press any key to continue.'
-
+  # kubectl rollout restart deployments --namespace ${INGRESS_NAMESPACE}
```

Comment:
Either remove this or use it. In general I would strongly prefer if we can use a proper `rollout restart`. The only issue I saw with this for Nginx was due to unavailable ports on the host, but that can be fixed with a `controller.updateStrategy.rollingUpdate.maxUnavailable=1` as I have noted in the readme.

Reply:
Good catch.